### PR TITLE
TINKERPOP-2491 Remove auto unfold of single item results for range, limit, tail local scope steps

### PR DESCRIPTION
**Pull request by andreachild**
[TINKERPOP-2491](https://issues.apache.org/jira/browse/TINKERPOP-2491)

Fixed inconsistent behaviour where `range()`, `limit()`, and `tail()` would return single elements instead of collections for single-item results with local scope:
- modified `RangeLocalStep.applyRangeIterable()` to always returns collections for iterable inputs regardless of number of results
- modified documentation and sample queries to reflect changed logic
- added more feature tests and unit tests

VOTE + 1

### CODE REVIEW

File: gremlin-test/src/main/resources/org/apache/tinkerpop/gremlin/test/features/filter/Range.feature [hunk 1]
```diff
@@ -191,7 +191,7 @@ Feature: Step - range()
       """
       g.V().fold().range(Scope.local, 6, 7)
       """
-    When iterated to list
+    When iterated next
```
Comment (xiazcy) [inline gremlin-test/src/main/resources/org/apache/tinkerpop/gremlin/test/features/filter/Range.feature]: Result of empty list can be represented as:
```
    When iterated to list
    Then the result should be unordered
      | result |
      | l[] |
```
I'd suggest either change this one or at least add another test for the empty list return.

**Comment by xiazcy** [comment]
VOTE+1 pending comment addressed.

File: docs/src/upgrade/release-3.8.x.asciidoc [hunk 2]
```diff
@@ -490,6 +490,52 @@ The `GremlinLangScriptEngine` has been modified to treat float literals without
 or 'd') as Double by default. Users who need `BigDecimal` precision can still use the 'm' suffix (e.g., 1.0m).
 `GremlinGroovyScriptEngine` will still default to `BigDecimal` for `float` literals.
 
+==== Consistent Collection Output for range(), limit(), and tail() Local Steps
```
Comment (spmallette) [inline docs/src/upgrade/release-3.8.x.asciidoc]: nit: needs a shorter section title or else it wraps terribly in the toc.

File: docs/src/upgrade/release-3.8.x.asciidoc [hunk 3]
```diff
@@ -490,6 +490,52 @@ The `GremlinLangScriptEngine` has been modified to treat float literals without
...
+
+gremlin> g.inject([1, 2, 3]).limit(local, 2)
+==>[1,2]  // collection returned
+
+// Map behavior unchanged in both versions
+gremlin> g.inject([a: 1, b: 2, c: 3]).limit(local, 1)
+==>[a:1]  // Map entry returned (behavior unchanged)
+----
+
+If you need the old behavior of extracting single elements from collections, you can add `.unfold()` after the local step:
```
Comment (spmallette) [inline docs/src/upgrade/release-3.8.x.asciidoc:551]: This semantic change is a troublesome one because folks won't see it as failing when the upgrade - their queries will just return new results. I think that warrants a much stronger statement here, where there is a marching order to actively review existing Gremlin for this pattern to include `unfold`.

**Comment by Cole-Greer** [comment]
VOTE +1 (pending resolution of above comments)

File: gremlin-test/src/main/resources/org/apache/tinkerpop/gremlin/test/features/filter/Range.feature (hunk 1 above)
Comment (andreachild) [reply gremlin-test/src/main/resources/org/apache/tinkerpop/gremlin/test/features/filter/Range.feature]: Changed as suggested

File: docs/src/upgrade/release-3.8.x.asciidoc (hunk 2 above)
Comment (andreachild) [reply docs/src/upgrade/release-3.8.x.asciidoc]: Shortened to 'Consistent Output for range(), limit(), tail()'

File: docs/src/upgrade/release-3.8.x.asciidoc (hunk 3 above)
Comment (andreachild) [reply docs/src/upgrade/release-3.8.x.asciidoc:551]: Added warning text area to specifically call out migration of existing queries that use the affected steps otherwise a class cast exception could occur or unpredictable results.
